presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/DigestAndPercentileArrayStateSerializer.java

```diff
@@ -44,18 +45,19 @@ public void serialize(DigestAndPercentileArrayState state, BlockBuilder out)
             out.appendNull();
         }
         else {
-            DynamicSliceOutput sliceOutput =
-                    new DynamicSliceOutput(state.getDigest().estimatedSerializedSizeInBytes() + (state.getPercentiles().size() * SIZE_OF_DOUBLE) +  SIZE_OF_LONG);
-            // write digest
-            state.getDigest().serialize(sliceOutput);
+            Slice digest = state.getDigest().serialize();
+
+            DynamicSliceOutput output = new DynamicSliceOutput(SIZE_OF_LONG + state.getPercentiles().size() * SIZE_OF_DOUBLE + digest.length());
             // write percentiles
             List<Double> percentiles = state.getPercentiles();
-            sliceOutput.appendLong(percentiles.size());
+            output.appendLong(percentiles.size());
             for (Double percentile : percentiles) {
-                sliceOutput.appendDouble(percentile);
+                output.appendDouble(percentile);
             }
 
-            VARBINARY.writeSlice(out, sliceOutput.slice());
+            output.appendBytes(digest);
```

Comment:
So in this serializer you don't write length and depend on the size of the overall slice... in the next one you write length.  Either one is fine with me, but you should choose one technique.